architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+
+#### User story 5
+
+As a user, upon the successful deployment of my application to my ACI environment, I can view the ACI containers in the CLI using the rad app graph command or via the Application Graph in the Radius Dashboard.
+
+#### User story 6
+
+As a user, I may set ACI-specific configurations for my containers via the extensions or runtimes properties. This allows me to make use of ACI-specific features like confidential containers or spot instances that may not be available on other container compute platforms. Note that any properties that are not applicable to the compute platform in the targeted deployment environment will be ignored.
+
+## User Experience (if applicable)
+
+The integration of Azure Container Instances (ACI) with Radius will bring several changes to the user experience, particularly in how users define and deploy their applications. The key changes are as follows:
+
+1. Environment Setup: Users will now be able to set up a Radius Environment targeted at ACI using environment definition files (e.g., env.bicep). This will involve specifying ACI-specific configurations and parameters in the Bicep templates.
+1. Application Definition: Users will define their applications using application definition files (e.g., app.bicep). These files can include ACI-specific properties and configurations, also know as “punch-through”, allowing users to tailor their deployments to the ACI environment.
+1. Deployment Process: The deployment process will have validation to ensure that when deploying to ACI users specify only configurations and properties that are valid for ACI.
+1. Container Definitions: The container definitions in the Bicep templates will be platform-agnostic, ensuring that users can deploy their containers to both Kubernetes and ACI without significant changes. However, users will have the option to apply ACI-specific configurations using extension properties in the Bicep templates.
+
+### Sample Input:Radius Environment*
+
+```bicep
+param aciscope string = '/subscriptions/<subscription id/resourceGroups/<resource group name>'
+resource env 'Applications.Core/environments@2023-10-01-preview' = {
+  name: 'aci-env'
+  properties: {
+    compute: {
+      kind: 'aci'
+      resourceGroup: aciscope
+    }
+    providers: {
+      azure: {
+        scope: aciscope
+      }
+    }
+  }
+}
+```
+
+### Sample Input: Portable Resources
+
+[Portable Resources](https://docs.radapp.io/guides/author-apps/portable-resources/overview/) are expected to work on ACI with no changes from the way they would be configured for Kubernetes:
```

Comment:
Will the connection from Radius Container to portable resource work the same way as well between Kubernetes and ACI?